packages/deployments/contracts/contracts/core/connext/helpers/SponsorVault.sol

```diff
@@ -316,4 +343,35 @@ contract SponsorVault is ISponsorVault, ReentrancyGuard, Ownable {
 
     connext = _connext;
   }
+
+  /**
+   * @notice Compare dex spot price and ChainLink oracle price to prevent manipulate
+   * @dev First fetch the price of the Native token and liquidity fee token from chainlink oracle.
+   *      If the aggregator of them not exist, return `false`
+   *      Compare (native_price_oracle / token_price_oracle) ~ (token_amount / eth_amount) = (native_price_dex / token_price_dex)
+   * @param _token The ERC20 token address of liquidity fee
+   * @param _ethAmount The amount of native token which got from `getInGivenExpectedOut`
+   * @param _tokenAmount The amount of liquidity fee to reimburse
+   */
+  function _checkDexSpotPrice(
+    address _token,
+    uint256 _ethAmount,
+    uint256 _tokenAmount
+  ) internal view returns (bool) {
+    uint256 ethPrice = priceOracle.getPriceFromChainlink(address(0));
+    if (ethPrice == 0) {
+      return false;
+    }
+
+    uint256 tokenPrice = priceOracle.getPriceFromChainlink(_token);
+    if (tokenPrice == 0) {
+      return false;
+    }
```

Comment:
Doesn't the `IPriceOracle` default to using chainlink -> oracle -> dex? maybe we could have a way to enforce the price feed source is acceptable instead of forcing it to come only from chainlink

Reply:
open to other arguments though! cc @wanglonghong

Reply:
> Doesn't the `IPriceOracle` default to using chainlink -> oracle -> dex? maybe we could have a way to enforce the price feed source is acceptable instead of forcing it to come only from chainlink

Our `ConnextPriceOracle` has a member function to get price feed from chainlink. `getPriceFromChainlink`